src/base-apis.js

```diff
@@ -49,2391 +49,1979 @@ function termsUrlForService(serviceType, baseUrl) {
- * @return {string} Identity Server URL of this client
  */
-MatrixBaseApis.prototype.getIdentityServerUrl = function(stripProto=false) {
-    if (stripProto && (this.idBaseUrl.startsWith("http://") ||
-            this.idBaseUrl.startsWith("https://"))) {
-        return this.idBaseUrl.split("://")[1];
+export class MatrixBaseApis {
+    /**
+     * @constructor
+     *
+     * @param {Object} opts Configuration options
+     *
+     * @param {string} opts.baseUrl Required. The base URL to the client-server
+     * HTTP API.
+     *
+     * @param {string} opts.idBaseUrl Optional. The base identity server URL for
+     * identity server requests.
+     *
+     * @param {Function} opts.request Required. The function to invoke for HTTP
+     * requests. The value of this property is typically <code>require("request")
+     * </code> as it returns a function which meets the required interface. See
+     * {@link requestFunction} for more information.
+     *
+     * @param {string} opts.accessToken The access_token for this user.
+     *
+     * @param {IdentityServerProvider} [opts.identityServer]
+     * Optional. A provider object with one function `getAccessToken`, which is a
+     * callback that returns a Promise<String> of an identity access token to supply
+     * with identity requests. If the object is unset, no access token will be
+     * supplied.
+     * See also https://github.com/vector-im/riot-web/issues/10615 which seeks to
+     * replace the previous approach of manual access tokens params with this
+     * callback throughout the SDK.
+     *
+     * @param {Number=} opts.localTimeoutMs Optional. The default maximum amount of
+     * time to wait before timing out HTTP requests. If not specified, there is no
+     * timeout.
+     *
+     * @param {Object} opts.queryParams Optional. Extra query parameters to append
+     * to all requests with this client. Useful for application services which require
+     * <code>?user_id=</code>.
+     *
+     * @param {boolean} [opts.useAuthorizationHeader = false] Set to true to use
+     * Authorization header instead of query param to send the access token to the server.
+     */
+    constructor(opts) {
+        utils.checkObjectHasKeys(opts, ["baseUrl", "request"]);
+        this.baseUrl = opts.baseUrl;
+        this.idBaseUrl = opts.idBaseUrl;
+        this.identityServer = opts.identityServer;
+        const httpOpts = {
+            baseUrl: opts.baseUrl,
+            idBaseUrl: opts.idBaseUrl,
+            accessToken: opts.accessToken,
+            request: opts.request,
+            prefix: PREFIX_R0,
+            onlyData: true,
+            extraParams: opts.queryParams,
+            localTimeoutMs: opts.localTimeoutMs,
+            useAuthorizationHeader: opts.useAuthorizationHeader,
+        };
+        this._http = new MatrixHttpApi(this, httpOpts);
+        this._txnCtr = 0;
     }
-    return this.idBaseUrl;
-};
-
```

Comment:
In general across the whole PR we lost a bunch of whitespace like this. Is it possible to get it back?

Reply:
um.. I'm sorry, lost what whitespace?

Reply:
the empty lines between functions, definitions within functions, etc.

Reply:
🤔does it necessary? (yes, I can try to make diff as small as possible)

Reply:
and since there is a big change going on, there is no need to review this pr any more. when the big change is done, @me and I will redo this pr

Reply:
well, it also removed it from constructors and stuff, so yes it is important. We also had it before and it's for our style preferences. Ideally changes like ES5 -> ES6 are purely mechanical and do not change the style in the slightest. There's also cases where brackets have been removed, and they shouldn't have (even if they do nothing).

Reply:
It would be best to regenerate whatever commits touched the whitespace, as otherwise all of the whitespace everywhere will show this PR in the annotate view even though it's unchanged.

Reply:
I'll close this pr and redo it. I'll open pr for each file so the maintainer can review it (and merge) it separately.

Reply:
please just put them all into one giant PR. It's easier to manage that way.